pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/ColumnMetadata.java

```diff
@@ -93,4 +94,6 @@ default boolean isMinMaxValueInvalid() {
   Map<IndexType<?, ?, ?>, Long> getIndexSizeMap();
 
   boolean isAutoGenerated();
+
+  List<Object> getChildColumns();
```

Comment:
Please add some javadoc explaining what is child column. Same for other interface change